internal/services/resource/parse/generic_resource.go

```diff
@@ -0,0 +1,55 @@
+package parse
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/resourceids"
+)
+
+// it's only to parse the existing resource.
+
+type GenericResourceId struct {
+	SubscriptionId   string
+	ResourceProvider string
+	ResourceType     string
+}
+
+// GenericResourceID parses a Generic Resource ID into an GenericResourceId struct
+func GenericResourceID(input string) (*GenericResourceId, error) {
+	id, err := resourceids.ParseAzureResourceID(input) //azure.ParseAzureResourceID(input)
+
+	if err != nil {
+		return nil, err
+	}
+
+	resourceTypeBuilder := strings.Builder{}
+	addSlash := false
+	for k, _ := range id.Path {
+		if addSlash {
+			resourceTypeBuilder.WriteString("/")
+		}
+		resourceTypeBuilder.WriteString(k)
+		addSlash = true
+	}
```

Comment:
There are cases that a resource id have more than one RPs, we should look into the last RP as its resource provider and resource type. But I think this is not the right place for this kind of code, probably the similar logic has been implemented in go-azure-helpers or go-azure-sdk?